src/api/graphql.ts

```diff
@@ -189,25 +208,19 @@ export async function getWalletInteractionsForContract(
       const parsedInput = inputTag?.value
         ? JSON.parse(inputTag.value)
         : undefined;
-      const sortKey = await interactionSorter.createSortKey(
-        e.node.block.id,
-        e.node.id,
-        e.node.block.height,
-      );
-      interactions.set(e.node.id, {
-        height: e.node.block.height,
-        timestamp: e.node.block.timestamp,
-        sortKey,
+      interactions.set(i.node.id, {
+        height: i.node.block.height,
+        timestamp: i.node.block.timestamp,
         input: parsedInput,
-        owner: e.node.owner.address,
+        owner: i.node.owner.address,
+        sortKey: i.node.sortKey,
       });
     }
     cursor =
-      response.data.data.transactions.edges[MAX_REQUEST_SIZE - 1]?.cursor ??
-      undefined;
-    hasNextPage =
-      response.data.data.transactions.pageInfo?.hasNextPage ?? false;
+      data.data.transactions.edges[MAX_REQUEST_SIZE - 1]?.cursor ?? undefined;
+    hasNextPage = data.data.transactions.pageInfo?.hasNextPage ?? false;
   } while (hasNextPage);
+
   return {
     interactions,
```

Comment:
Something to think about - this list will eventually become untenably large and we may have to start thinking about how to handle these types of requests in a streaming fashion (e.g. some combination of generator functions, streamed responses, and pages of data at a time).

Reply:
we have a ticket to add pagination, but there is more work to do for stream responses/generators - will create a ticket

Reply:
Pagination is great start. There isn't yet a clear use case for a streaming version and the complexity would be way higher. You're on the right track already.